gtsam/geometry/SphericalCamera.h

```diff
@@ -0,0 +1,217 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file SphericalCamera.h
+ * @brief Calibrated camera with spherical projection
+ * @date Aug 26, 2021
+ * @author Luca Carlone
+ */
+
+#pragma once
+
+#include <gtsam/base/Manifold.h>
+#include <gtsam/base/ThreadsafeException.h>
+#include <gtsam/base/concepts.h>
+#include <gtsam/dllexport.h>
+#include <gtsam/geometry/BearingRange.h>
+#include <gtsam/geometry/Pose3.h>
+#include <gtsam/geometry/Unit3.h>
+
+#include <boost/serialization/nvp.hpp>
+
+namespace gtsam {
+
+class GTSAM_EXPORT EmptyCal {
+ public:
+  enum { dimension = 0 };
+  EmptyCal() {}
+  virtual ~EmptyCal() = default;
+  using shared_ptr = boost::shared_ptr<EmptyCal>;
+  void print(const std::string& s) const {
+    std::cout << "empty calibration: " << s << std::endl;
+  }
+};
+
+/**
+ * A spherical camera class that has a Pose3 and measures bearing vectors
```

Comment:
Note here that it has 
‘Empty’ calibration and the only 6 degrees of freedom are the pose

Reply:
done

Reply:
done